File: examples/csharp/Send.cs
using System.Net.Http.Headers;
using System.Text;

var apiUrl = Environment.GetEnvironmentVariable("NEXTMSG_URL") ?? "http://localhost:4000";
var token = Environment.GetEnvironmentVariable("NEXTMSG_TOKEN")!;
var instanceId = Environment.GetEnvironmentVariable("NEXTMSG_INSTANCE_ID")!;

using var client = new HttpClient();
client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);

var json = $$"""
{
  "instanceId": "{{instanceId}}",
  "to": "+15551234567",
  "body": "Hello from C#"
}
""";

var response = await client.PostAsync(
    $"{apiUrl}/messages/send",
    new StringContent(json, Encoding.UTF8, "application/json")
);

response.EnsureSuccessStatusCode();
Console.WriteLine(await response.Content.ReadAsStringAsync());

File: examples/php/send.php
<?php
$apiUrl = getenv('NEXTMSG_URL') ?: 'http://localhost:4000';
$token = getenv('NEXTMSG_TOKEN');
$instanceId = getenv('NEXTMSG_INSTANCE_ID');

$payload = json_encode([
    'instanceId' => $instanceId,
    'to' => '+15551234567',
    'body' => 'Hello from PHP',
]);

$ch = curl_init("$apiUrl/messages/send");
curl_setopt_array($ch, [
    CURLOPT_POST => true,
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_HTTPHEADER => [
        "Authorization: Bearer $token",
        "Content-Type: application/json",
    ],
    CURLOPT_POSTFIELDS => $payload,
]);

$body = curl_exec($ch);
if ($body === false) {
    throw new RuntimeException(curl_error($ch));
}
curl_close($ch);
echo $body . PHP_EOL;
